src/components/Reference.js

```diff
@@ -0,0 +1,8 @@
+import pure from 'rc-pure-component';
+
+const Reference = ({ children }) => children;
+
+const Wrapper = pure(Reference);
+Wrapper.displayName = 'Wrapper';
```

Comment:
React has a built-in method for this - `PureComponent`  https://reactjs.org/docs/react-api.html#reactpurecomponent
Starting from v.16.6 React has `React.memo` https://reactjs.org/docs/react-api.html#reactmemo

I don't think it's a good idea to use such new methods, so let's use PureComponent for now
```javascript
import React, { PureComponent } from 'react';
export default class Reference extends PureComponent {
  render() {
    return this.props.children;
  }
}
```

and later we can replace this with
```javascript
import React from 'react';
export default React.memo(({ children }) => children)
```

------

and, as I suggested earlier, if we pass `undefined` as a child here - everything should work, right?

Reply:
I removed it!